File: Capestone_ui/CapeStoneProject/CapeStoneProject/src/main/java/com/capestone/model/Officer.java
package com.capestone.model;

import com.capestone.enums.JobTitle;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Entity
@Getter
@Setter
public class Officer {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    private String name;

    @Enumerated(EnumType.STRING)
    private JobTitle jobTitle;

    @OneToOne
    private User user;
}

File: python_project/models/officer.py
class Officer:
    def __init__(self, officer_id, name, job_title, user_id):
        self.__id = officer_id
        self.__name = name
        self.__job_title = job_title
        self.__user_id = user_id

    @property
    def id(self):
        return self.__id

    @id.setter
    def id(self, value):
        self.__id = value

    @property
    def name(self):
        return self.__name

    @name.setter
    def name(self, value):
        self.__name = value

    @property
    def job_title(self):
        return self.__job_title

    @job_title.setter
    def job_title(self, value):
        self.__job_title = value

    @property
    def user_id(self):
        return self.__user_id

    @user_id.setter
    def user_id(self, value):
        self.__user_id = value
